api/v1alpha/src/admin/dataManager.js

```diff
@@ -228,8 +229,82 @@ async function setupDatasharePrerequisites(projectId) {
     });
 }
 
+/**
+ * Initializes PubSub listener for entitlement auto approvals
+ */
+async function initializePubSubListiner() {
+    console.log(`Initializing PubSub listener`);
+
+    let projectId = '';
+    // https://github.com/googleapis/gcp-metadata
+    // https://cloud.google.com/appengine/docs/standard/java/accessing-instance-metadata
+    const gcpMetadata = require('gcp-metadata');
+    const isAvailable = await gcpMetadata.isAvailable();
+    if (isAvailable === true) {
+        console.log('gcpMetadata is available, getting projectId');
+        projectId = await gcpMetadata.project('project-id');
+        console.log(projectId); // ...Project ID of the running instance
+    } else {
+        console.log('gcpMetadata is unavailable, will not start PubSub listener');
+
+        if (process.env.NODE_ENV === 'production' || process.env.VUE_APP_APICLIENT == 'server') {
+            console.log('Could not identify project, will not start up subscription');
+            return;
+        } else {
+            console.log('Running locally will default to demo project');
+            projectId = 'cds-demo-2';
+        }
+    }
+
+    const topicName = `projects/cloudcommerceproc-prod/topics/${projectId}`;
+    const subscriptionName = `projects/${projectId}/subscriptions/procurement-${projectId}`;
+    const { PubSub } = require('@google-cloud/pubsub');
+    const pubSubClient = new PubSub();
+
+    const [subscriptions] = await pubSubClient.getSubscriptions();
+    const subscription = underscore.findWhere(subscriptions.flatMap(e => e.metadata), { name: subscriptionName, topic: topicName });
+
+    if (subscription) {
+        console.log(`Subscription '${subscriptionName}' already exists`)
+    } else {
+        await pubSubClient.topic(topicName).createSubscription(subscriptionName);
+        console.log(`Subscription '${subscriptionName}' created.`);
+    }
```

Comment:
Should these be added to the shared [pubSubUtil](https://github.com/GoogleCloudPlatform/datashare-toolkit/blob/master/shared/pubSubUtil.js#L92) library instead?